include/sparrow/array/external_array_data.hpp

```diff
@@ -96,6 +96,7 @@ namespace sparrow
     {
     public:
 
+        static constexpr bool is_mutable = false;
```

Comment:
To complete this addition, it would be best to also add a requirement in the [`data_storage`](https://github.com/man-group/sparrow/blob/main/include/sparrow/array/array_data_concepts.hpp#L59-L75) for this member to exist and be convertible to bool, that way we're sure future breaking changes will get caught even before using these types in layout impls.